mpas_analysis/shared/io/namelist_streams_interface.py

```diff
@@ -0,0 +1,91 @@
+#!/usr/bin/python
+"""
+    Module of classes / routines to manipulate fortran namelist and streams
+    files.
+
+    Phillip Wolfram
+    LANL
+    04/24/2014
+"""
+
+from subprocess import check_output, call
+from lxml import etree
+
+class NameList:
+    """
+    Class for fortran manipulation of namelist files, provides
+    read and write functionality
+    """
+
+    # constructor
+    def __init__(self, fname, readonly=True):
+        # set readonly bit
+        self.readonly = readonly
+        # input file name
+        self.fname = fname
+
+    def read(self, name):
+        # shell return value
+        return_val = check_output(['awk', '/'+name+'/{printf $3}', self.fname])
+        return_val.rstrip('\n')
+        try:
+            return_val = float(return_val)
+        except:
+            return_val = return_val.strip('"').strip("'")
+        return return_val
+
+    def write(self, name, value):
```

Comment:
@xylar, we may want to use this type of code to make edits for namelists / streams for automatic test cases.  This was the context that this code was original written to address.  I think there is an advantage to having this functionality, even if we don't plan to immediately use it because we want the O part of IO too in order to make this a general tool.  For example, I can easily envision using MPAS-Analysis to setup/analyze test cases in the testing core and this would be useful in this endeavor.